### PR TITLE
Use many linux image in several places

### PR DESCRIPTION
**Pull request by samansmink**
This PR is basically what @hannes PR-ed in https://github.com/duckdb/duckdb/pull/8028 with some extras

## This PR
- `centos_7_setup` action renamed to `manylinux_2014_setup` for clarity
- switch custom openssl builds vcpkg global installation
- following jobs now also use manylinux:
	- linux-extensions-64-rebuild
	- jdbc-compliance
	- linux-release-64
	- linux-extensions-64
	- check-load-install-extensions
	- linux-release-32
- linux-rpi was removed
- linux-release-32 release disabled, we only run the tests now. Also, it now uses the GCC4.8 compiler (installing the cross compiler on manylinux was pain)

### aarch64 linux & extensions
This is a bit annoying to get right on GH actions. Ideally we would just just use the aarch64 manylinux2014 image. However, GH actions doesnt have aarch64 runners. Also, GH actions doesn't support docker with multi-arch. And even if it would work, I think it would be super slow.

So basically, we could do manylinux binaries for aarch64 by manually installing qemu, then running the build in docker. This would require some more refactoring to the build_extensions action essentially breaking that up into separate parts run as scripts that could be used between the regular x86 build in github actions and the aarch64 build in docker. Alternatively we could use the x86 manylinux image to cross compile, but getting that to work feels tricky and fragile. Finally we could also just leave it to the ubuntu 18 image we currently use. 

### regression tests in manylinux
II think we can just move these to manylinux too, will do so in separate PR later

### CODE REVIEW

**Comment by samansmink** [comment]
CI failure seems unrelated: https://github.com/duckdb/duckdb/actions/runs/5410029559/jobs/9832590997?pr=8097

**Comment by Mytherin** [comment]
Thanks! Looks great!
